src/Mailer/EmailTemplateFactory.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace AppBundle\Mailer;
+
+use AppBundle\Mailer\Message\Message;
+
+class EmailTemplateFactory
+{
+    private $senderEmail;
+    private $senderName;
+    private $templateClass;
+
+    public function __construct(string $senderEmail, string $senderName, string $templateClass)
+    {
+        $this->senderEmail = $senderEmail;
+        $this->senderName = $senderName;
+        $this->templateClass = $templateClass;
+    }
+
+    public function createFromMessage(Message $message): EmailTemplate
+    {
+        return call_user_func_array([$this->templateClass, 'createWithMessage'], [
```

Comment:
We should add a test before to ensure the method exists in the class or throw an exception instead.